catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/app/PenColorBrickTest.java

```diff
@@ -0,0 +1,225 @@
+		onView(withId(R.id.color_rgb_seekbar_blue))
+				.perform(CustomSwipeAction.swipeToPosition(setRgbValuesBlue[0] / 255.0f));
+		onView(withId(R.id.color_rgb_seekbar_green))
+				.perform(CustomSwipeAction.swipeToPosition(setRgbValuesGreen[1] / 255.0f));
+		onView(withId(R.id.color_rgb_seekbar_green))
+				.perform(CustomSwipeAction.swipeToPosition(setRgbValuesGreen[2] / 255.0f));
+		onView(withId(R.id.color_rgb_seekbar_blue))
+				.perform(CustomSwipeAction.swipeToPosition(setRgbValuesBlue[1] / 255.0f));
+		onView(withId(R.id.color_rgb_seekbar_red))
+				.perform(CustomSwipeAction.swipeToPosition(setRgbValuesRed[1] / 255.0f));
+		onView(withId(R.id.color_rgb_seekbar_red))
+				.perform(CustomSwipeAction.swipeToPosition(setRgbValuesRed[2]));
+
+		onView(withId(R.id.menu_undo)).perform(click());
+		onView(withId(R.id.menu_undo)).perform(click());
+		onView(withId(R.id.menu_undo)).perform(click());
+		onView(withId(R.id.menu_undo)).perform(click());
+
+		onView(withId(R.id.menu_redo)).perform(click());
+		onView(withId(R.id.menu_redo)).perform(click());
+		onView(withId(R.id.menu_redo)).perform(click());
+
+		onView(withId(R.id.color_rgb_seekbar_red))
+				.perform(pressBack());
+		onView(withText(R.string.formula_editor_discard_changes_dialog_title))
+				.check(matches(isDisplayed()));
+		onView(withId(android.R.id.button1))
+				.perform(click());
+		onBrickAtPosition(positionPenColorBrick).onFormulaTextField(R.id.brick_set_pen_color_action_red_edit_text)
+				.checkShowsNumber((int) setRgbValuesRed[1]);
+		onBrickAtPosition(positionPenColorBrick).onFormulaTextField(R.id.brick_set_pen_color_action_green_edit_text)
+				.checkShowsNumber((int) setRgbValuesGreen[2]);
+		onBrickAtPosition(positionPenColorBrick).onFormulaTextField(R.id.brick_set_pen_color_action_blue_edit_text)
+				.checkShowsNumber((int) setRgbValuesBlue[1]);
+	}
+
+	@Category({Cat.AppUi.class, Level.Smoke.class})
+	@Test
+	public void testPenColorBrickDiscardChanges() {
+		int positionPenColorBrick = 1;
+		float[] newRgbValues = {55, 122, 99};
+
+		onBrickAtPosition(positionPenColorBrick).onFormulaTextField(R.id.brick_set_pen_color_action_red_edit_text)
+				.perform(click());
+		onView(withId(R.id.color_rgb_seekbar_red))
+				.perform(CustomSwipeAction.swipeToPosition(newRgbValues[0] / 255.0f));
+		onView(withId(R.id.color_rgb_seekbar_green))
+				.perform(CustomSwipeAction.swipeToPosition(newRgbValues[1] / 255.0f));
+		onView(withId(R.id.color_rgb_seekbar_blue))
+				.perform(CustomSwipeAction.swipeToPosition(newRgbValues[2] / 255.0f));
+		onView(withId(R.id.color_rgb_seekbar_red))
+				.perform(pressBack());
+		onView(withText(R.string.formula_editor_discard_changes_dialog_title))
+				.check(matches(isDisplayed()));
+		onView(withId(android.R.id.button2))
+				.perform(click());
+		onBrickAtPosition(positionPenColorBrick).onFormulaTextField(R.id.brick_set_pen_color_action_red_edit_text)
+				.checkShowsNumber(255);
```

Comment:
please make these variables eg. INITIAL_BLUE, INITIAL_RED, etc...

Reply:
please write a ViewInteractionWrapper wrapping up the whole color picker view. have a look at FormualEditorViewInteractionWrapper.

for the color picker please put this in a separate class since were probably gonna need it more often, and for the phiro one you could just add an inline class:
		onSeekbar().performEnterValue(4)
				.performEnterValue(4)
				.performEnterValue(34)
				.check(matches(withText("234")));
	}

	private static class PickerWrapper extends ViewInteractionWrapper {
		static final Matcher<View> SEEKBAR = withId(R.id.single_seekbar_seekbar);
		private PickerWrapper() {
			super(onView(SEEKBAR));
		}

		static PickerWrapper onSeekbar() {
			return new PickerWrapper();
		}

		PickerWrapper performEnterValue(float value) {
			viewInteraction.perform(new CustomSwipeAction(Swipe.SLOW, CustomSwipeAction.SwipeAction.SWIPE_RIGHT,
					Press.FINGER));
			return this;
		}
		
		PickerWrapper checkMatchesValue(float value) {
			...
		}
	}